### PR TITLE
Add init_batch_args to SampleQC stage query_command() and Refactor Reusability Checks

### PR DESCRIPTION
**Pull request by michael-harper**
This PR refactors allows additional customisation and initialisation options for the SampleQC batch process to provide more worker and driver memory as well as more driver cores.

This is done via providing `init_batch_args` to `query_command()`.

### CODE REVIEW

**Review by Alexander-Stuckey — APPROVED** [review]
Looks good to me

File: cpg_workflows/large_cohort/sample_qc.py [hunk 1]
```diff
@@ -15,7 +15,27 @@
...
+        return False
+
+    path_as_path = to_path(path)
+
+    if path_as_path.suffix in ['.mt', '.ht']:
+        path_as_path /= '_SUCCESS'
+    if path_as_path.suffix in ['.vds']:
+        path_as_path /= 'variant_data/_SUCCESS'
+
+    return path_as_path.exists()
```
Comment (MattWellie) [inline cpg_workflows/large_cohort/sample_qc.py:34]: this should be reused from the cpg_workflows module

Comment (MattWellie) [reply cpg_workflows/large_cohort/sample_qc.py:34]:
```
from cpg_workflows.utils import can_reuse
```

**Review by MattWellie — CHANGES_REQUESTED** [review]

File: cpg_workflows/large_cohort/sample_qc.py (hunk 1 above)
Comment (MattWellie) [reply cpg_workflows/large_cohort/sample_qc.py:34]: can_reuse calls exists, exists calls exists_not_cached, exists_not_cached does this `_SUCCESS` check
https://github.com/populationgenomics/production-pipelines/blob/d37ca03a06bb9b8c0df22ea8ac13d3ed1d5de797/cpg_workflows/utils.py#L181

**Comment by michael-harper** [comment]
Thanks @MattWellie ! I've removed the `can_resuse_dataset` function and gone back to the imported `can_reuse`. I added a logging statement that tells use if it's reusing the checkpointed vds.

File: cpg_workflows/large_cohort/sample_qc.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 
 def run(vds_path: str, out_sample_qc_ht_path: str, tmp_prefix: str):
+    logging.basicConfig(level=logging.INFO)
     if can_reuse(out_sample_qc_ht_path):
         return []
```
Comment (Alexander-Stuckey) [inline cpg_workflows/large_cohort/sample_qc.py:21]: If we can reuse the output, shouldn't we return the path?

**Review by Alexander-Stuckey — COMMENTED** [review]
Just one comment

**Review by MattWellie — APPROVED** [review]
LGTMatt

**Pull request merged**